### ISSUE TITLE
[Linux] Touchscreen issue, cursor jumping after movement

### ISSUE THREAD

**Issue by hippydave** [alexbatalov/fallout1-ce#172]
I know the readme only talks about touchscreen controls for Android/iOS, but it appears to be active in the Linux build too. I'm trying to run the game on L4T Ubuntu on my Nintendo Switch (using the Linux x64 build, via box64 to run on the Switch's AArch64 architecture). It seems to run totally fine, except when I get to the main menu I'm having an issue controlling the mouse cursor with the touchscreen.

While I move my finger on the screen, the cursor moves as expected, doing the same relative movements. But when I lift my finger from the screen, the cursor does a sudden jump. If I've just moved in one direction, the cursor will jump in the same direction, and the further I've moved, the further it jumps. If I've moved in multiple directions in the same touch, then the jump seems to be in the final direction I was moving, but a jump size related to the total distance moved.

One- and two-finger taps for left and right mouse button seem to work as expected, so the cursor jumping after movement is the only thing keeping me from playing. Does anyone have any suggestions please? Many thanks!

**Comment by theofficialgman** [comment]
It should be possible to build this from source rather than use the provided x64 Linux builds with box64. You will get better performance and less power draw.

**Comment by hippydave** [comment]
> It should be possible to build this from source rather than use the provided x64 Linux builds with box64. You will get better performance and less power draw.

Yep was thinking about it, but I don't currently have a Linux build environment set up, and there's not much spare space on my Switch's Linux partition. Unless I could build it on my Raspberry Pi maybe?

Haven't spent much time in game yet because of the aforementioned issue, but it seems like performance is ample anyway for this 1997 game. Would be nice to have it running native tho.

**Comment by hippydave** [comment]
Not the focus of this issue, but for the glitchy videos, I tried [this commit](https://github.com/Zenkibou/fallout1-ce/commit/deb3f21c7f9f03616b666e748be40936b59686db) which says it fixes it, but didn't work for me so I tried rolling back [this previous fix](https://github.com/hippydave/fallout1-ce/commit/ad8a2754fa99ceaf40495185ad48191cf27559e2) which apparently introduced the regression and they're playing fine now.

**Comment by hippydave** [comment]
I identified the issue causing the cursor jumps, and made a [pull request](https://github.com/alexbatalov/fallout1-ce/pull/176) with a potential fix.

**Comment by Norfolc** [comment]
The same issue reproduces also on Windows laptop with touch screen. The proposed fix #176 resolves the issue.
